### PR TITLE
16 create cache algorithm

### PR DESCRIPTION
**Pull request by amalikrigger**
Making a few changes to NetworkManager and adding in cache functionality for SearchResults and Images

### CODE REVIEW

**Review by Denoble — COMMENTED** [review]
Lawrence has  similar function for loadimage function as well in the view controller. We can have one function only for this.

File: CandySpaceTests/CandySpaceTests.swift [hunk 1]
```diff
@@ -9,28 +9,22 @@ import XCTest
...
-    // swiftlint:enable all
+     override func tearDown() {
+         super.tearDown()
+         viewModel = nil
+     }
+     func testGetImageGallery() async {
+         let expectedImages = 20
+         await viewModel!.getImageGallery(searchTerm: "nature")
+         XCTAssertEqual(viewModel?.images.count, expectedImages, "Hits are not equal")
+         await viewModel!.getImageGallery(searchTerm: "nature")
```
Comment (chiawei1103) [inline CandySpaceTests/CandySpaceTests.swift:27]: same as line 25?

File: CandySpaceTests/GalleryViewModelTests.swift
Comment (chiawei1103) [inline CandySpaceTests/GalleryViewModelTests.swift:1]: Is this file same as CandySpaceTests?

File: CandySpaceTests/CandySpaceTests.swift (hunk 1 above)
Comment (amalikrigger) [reply CandySpaceTests/CandySpaceTests.swift:27]: This checks the cache. Well it is supposed to

File: CandySpaceTests/GalleryViewModelTests.swift
Comment (amalikrigger) [reply CandySpaceTests/GalleryViewModelTests.swift:1]: Yes it is a duplicate. Not needed, but it doesn't hurt anything at the moment

**Review by Denoble — APPROVED** [review]

**Review by Wrozier — APPROVED** [review]
-looks good

**Review by chiawei1103 — APPROVED** [review]

**Pull request merged**